src/ed25519/curve.rs

```diff
@@ -357,8 +345,13 @@ impl CurveExt for Ed25519 {
 
     const CURVE_ID: &'static str = "ed25519";
 
+    fn is_on_curve(&self) -> Choice {
+        let affine = Ed25519Affine::from(*self);
+        !self.z.is_zero() & affine.is_on_curve() & (affine.x * affine.y * self.z).ct_eq(&self.t)
```

Comment:
does this mean the infinity point will return false?

Reply:
No. There's no infinity point in TE curves. The identity element is `(x, y) = (0, 1)` which in [extended coordinates](https://en.wikipedia.org/wiki/Twisted_Edwards_curve#Extended_coordinates) corresponds to `(x, y, z, t) = (0, 1, 1, 0)`

Reply:
right... my bad